docs/guide/pagination.md

```diff
@@ -12,20 +12,190 @@ An interface for limit/offset pagination can be use for basic pagination needs:
 @strawberry_django.type(models.Fruit, pagination=True)
 class Fruit:
     name: auto
+
+
+@strawberry.type
+class Query:
+    fruits: list[Fruit] = strawberry_django.field()
+```
+
+Would produce the following schema:
+
+```graphql title="schema.graphql"
+type Fruit {
+  name: String!
+}
+
+type Query {
+  fruits(pagination: PaginationInput): [Fruit!]!
+}
 ```
 
+And can be queried like:
+
 ```graphql title="schema.graphql"
 query {
   fruits(pagination: { offset: 0, limit: 2 }) {
     name
-    color
   }
 }
 ```
 
-There is not default limit defined. All elements are returned if no pagination limit is defined.
+The `pagination` argument can be given to the type, which will enforce the pagination
+argument every time the field is annotated as a list, but you can also give it directly
+to the field for more control, like:
+
+```python title="types.py"
+@strawberry_django.type(models.Fruit)
+class Fruit:
+    name: auto
+
+
+@strawberry.type
+class Query:
+    fruits: list[Fruit] = strawberry_django.field(pagination=True)
+```
+
+Which will produce the exact same schema.
+
+> [!NOTE]
+> There is no default limit defined. All elements are returned if no pagination limit is defined.
+
+## Paginated Generic
+
+For more complex pagination needs, you can use the `Paginated` generic, which alongside
+the `pagination` argument, will wrap the results in an object that contains the results
+and the pagination information, together with the `totalCount` of elements excluding pagination.
+
+```python title="types.py"
+from strawberry_django.pagination import Paginated
```

Comment:
I wonder if we should call this `OffsetPaginated` or something like that 🤔

Reply:
Decided to go with `OffsetPaginated`, to be aligned with the `OffsetPaginationInput` that we already have :)